demos/http/http_demo_s3_download_multithreaded/http_demo_s3_download_multithreaded.c

```diff
@@ -0,0 +1,1022 @@
+        {
+            /* Settings for constructing queues. */
+            struct mq_attr queueSettings;
+
+            queueSettings.mq_maxmsg = QUEUE_SIZE;
+            queueSettings.mq_msgsize = sizeof( RequestItem_t );
+
+            requestQueue = mq_open( REQUEST_QUEUE,
+
+                                    /* These options create a queue if it does
+                                     * not already exist, and then opens it in
+                                     * non-blocking mode. It is opened as
+                                     * write-only as the main thread only writes
+                                     * HTTP requests to it. */
+                                    O_CREAT | O_NONBLOCK | O_WRONLY,
+                                    QUEUE_PERMISSIONS,
+                                    &queueSettings );
+
+            queueSettings.mq_msgsize = sizeof( ResponseItem_t );
+
+            responseQueue = mq_open( RESPONSE_QUEUE,
+
+                                     /* These options create a queue if it does
+                                      * not already exist, and then opens it in
+                                      * non-blocking mode. It is opened as
+                                      * read-only as the main thread only reads
+                                      * HTTP responses from it. */
+                                     O_CREAT | O_NONBLOCK | O_RDONLY,
+                                     QUEUE_PERMISSIONS,
+                                     &queueSettings );
+
+            if( requestQueue == -1 )
+            {
+                LogError( ( "Failed to open request queue with error %s.",
+                            strerror( errno ) ) );
+                returnStatus = EXIT_FAILURE;
+            }
+
+            if( responseQueue == -1 )
+            {
+                LogError( ( "Failed to open response queue with error %s.",
+                            strerror( errno ) ) );
+                returnStatus = EXIT_FAILURE;
+            }
+        }
+
+        /* Start the HTTP task which services requests in requestQueue. */
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            httpThread = startHTTPThread( &transportInterface );
+
+            if( httpThread == -1 )
+            {
+                returnStatus = EXIT_SUCCESS;
+            }
+        }
+
+        /******************** Download S3 Object File. **********************/
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            bool result = false;
+            result = downloadS3ObjectFile( pHost,
+                                           hostLen,
+                                           pPath,
+                                           requestLen,
+                                           requestQueue,
+                                           responseQueue );
+
+            if( result == false )
+            {
+                returnStatus = EXIT_FAILURE;
+            }
+        }
```

Comment:
I think you will need to print "Demo completed successfully." in the event that it does for CI to be able to count a run as successful. You can sync up with @nateglims for that.

Reply:
Yes you will def need to print 
```
            /* Log a message indicating an iteration completed successfully. */
            LogInfo( ( "Demo completed successfully." ) );
```